java/arcs/core/storage/DirectStore.kt

```diff
@@ -60,15 +67,22 @@ class DirectStore<Data : CrdtData, Op : CrdtOperation, T> /* internal */ constru
     private var pendingDriverModels = atomic(listOf<PendingDriverModel<Data>>())
     private var version = atomic(0)
     private var state: AtomicRef<State<Data>> = atomic(State.Idle(idleDeferred, driver))
+    private val stateChannel =
+        ConflatedBroadcastChannel<State<Data>>(State.Idle(idleDeferred, driver))
+    private val stateFlow = stateChannel.asFlow()
     private val proxyManager = RandomProxyCallbackManager<Data, Op, T>(
         "direct",
         Random
     )
 
+    private val storeIdlenessFlow =
+        combine(stateFlow, writebackIdlenessFlow) { state, writebackIsIdle ->
```

Comment:
nice!

Reply:
https://pl.kotl.in/KMeqmenwZ here's an example of this type of thing in action